pkg/actuators/machine/instances.go

```diff
@@ -422,9 +427,27 @@ func buildTagList(machineName string, clusterID string, machineTags []awsprovide
 		{Key: aws.String("kubernetes.io/cluster/" + clusterID), Value: aws.String("owned")},
 		{Key: aws.String("Name"), Value: aws.String(machineName)},
 	}...)
+
 	return removeDuplicatedTags(rawTagList)
 }
 
+// mergeInfrastructureAndMachineSpecTags merge list of tags from machine provider spec and Infrastructure object platform spec.
+func mergeInfrastructureAndMachineSpecTags(machineSpecTags []awsproviderv1.TagSpecification, infra *configv1.Infrastructure) []awsproviderv1.TagSpecification {
+	if infra == nil || infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.AWS == nil || infra.Status.PlatformStatus.AWS.ResourceTags == nil {
+		return machineSpecTags
+	}
+
+	mergedList := []awsproviderv1.TagSpecification{}
+
+	for _, tag := range infra.Status.PlatformStatus.AWS.ResourceTags {
+		mergedList = append(mergedList, awsproviderv1.TagSpecification{Name: tag.Key, Value: tag.Value})
+	}
+
+	mergedList = append(mergedList, machineSpecTags...)
+
+	return mergedList
+}
+
```

Comment:
Which of these takes precedence? We updated the agreement to a local override, so we need to make sure that Machine tags override the Infra tags

Please make sure this is tested somewhere and that a comment is added to the function comment so we note this in the future